src/hal/uart/Uart.c

```diff
@@ -28,43 +28,47 @@ void (*uartHTTPReceive)(char *) = NULL;
 
 void (*uartInternalCallbackUartRxInterrupt)(void);
 
+static uartConfiguration_t uartConfig;
+
 /* endregion  VARIABLES*/
 
 /* region HEADER FUNCTION IMPLEMENTATIONS */
 
-void uartInit(uartConfiguration_t *uartConfig) {
+void uartInit(uartConfiguration_t *externalUartConfig) {
+
+    uartConfig = *externalUartConfig;
```

Comment:
```suggestion
void uartInit(uartConfiguration_t configuration) {
    uartConfig = configuration;
```
Due to the fact that we store a copy of the struct and not a pointer to the struct. I suggest to just pass a struct instead of a pointer.

Reply:
Agreed, or at least the pointer should be const in this case.